hrjobcontract/js/test/controllers/contract-list_test.js

```diff
@@ -0,0 +1,106 @@
+    var ctrl, $rootScope, $controller, $scope, $q, $httpBackend, ContractHealthService;
+
+    beforeEach(module('hrjc'));
+
+    beforeEach(module(function ($provide) {
+      $provide.factory('ContractHealthService', function() {
+        return {
+          getOptions: function () {},
+          getFields: function () {}
+        }
+      });
+    }));
+
+    beforeEach(inject(function (_$controller_, _$rootScope_, _$q_, _$httpBackend_, _ContractHealthService_) {
+      $controller = _$controller_;
+      $rootScope = _$rootScope_;
+      $httpBackend = _$httpBackend_;
+      $q = _$q_;
+      ContractHealthService = _ContractHealthService_;
+
+      contractHealthServiceSpy()
+
+      makeController();
+    }));
+
+    beforeEach(function () {
+      $httpBackend.whenGET(/action=get&entity=HRHoursLocation/).respond({});
+      $httpBackend.whenGET(/action=get&entity=HRPayScale/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobDetails/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobHour/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobPay/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobLeave/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobHealth/).respond({});
+      $httpBackend.whenGET(/action=getfields&entity=HRJobPension/).respond({});
+      $httpBackend.whenGET(/action=get&entity=HRJobContract/).respond({});
+      $httpBackend.whenGET(/action=getoptions&entity=HRJobHealth/).respond({});
+      $httpBackend.whenGET(/views.*/).respond({});
+    });
+
+    describe("Fetch Health Plans Options on opening new ontract modal", function () {
+      describe("when calling modalContract(params)", function() {
+        beforeEach(function() {
+          $rootScope.$digest();
+        });
+
+        beforeEach(function() {
+          var action = "new",
+            health = {}
+
+          health.plan_type = {}
+          $rootScope.options = {
+            health: health
+          };
+          $scope.modalContract(action);
+          $rootScope.$digest();
+        });
+
+        var result = {
+          FamilyOne: "FamilyOne",
+          Individual: "Individual"
+        }
+
+        it("calls getOptions() form ContractHealthService", function() {
+          expect(ContractHealthService.getOptions).toHaveBeenCalled();
+        });
+
+        it("fetches health plan types", function() {
+          expect($rootScope.options.health.plan_type).toEqual(result)
+        });
+      });
+    });
+
+    function makeController () {
+      $scope = $rootScope.$new();
+      ctrl = $controller('ContractListCtrl', {
+        $scope: $scope,
+        $rootScope: $rootScope,
+        contractList: []
+      });
+    }
+
+    function contractHealthServiceSpy () {
+      spyOn(ContractHealthService, "getOptions").and.callFake(function () {
+        var deferred = $q.defer();
+        deferred.resolve([
+          {
```

Comment:
Create a `js/test/mock/data` folder in which you will put this mocked data, then you can then just require in this file. Please make the mocked data as "belieavable" as it can, mocking the exact structure of what you would get from the api